### ISSUE TITLE
Renegade: sprite RAM length

### ISSUE THREAD

**Issue by jotego** [mamedev/mame#10163]
We are working on the Renegade PCB schematics. We have not finished yet, but it called my attention that what it seemed like the sprite RAM was only connected to bits 7:0 of the address bus. Bits 6:0 connect directly, and bit 7 selects the upper or lower module. This means that the total connected memory is only 256 bytes, and not 2kB as the driver has. I have confirmed this by reducing the memory to 256 and still passing the built-in memory test.

![imagen](https://user-images.githubusercontent.com/1863036/182043101-a9f1a544-954b-4377-897f-fbac0d17e058.png)

I think this line should address only 256 bytes and mirror the rest of the space.

https://github.com/mamedev/mame/blob/0fa347410774648b55b4e0af508098a6fbd1902d/src/mame/technos/renegade.cpp#L273

This also means that the sprite table is not 96-sprite long, but only 64 (as 64*4=256).

**Comment by cuavas** [comment]
I’ll take care of this one.

**Comment by jotego** [comment]
@cuavas, sorry I rushed too much in my post. Although the test passed with just 256 bytes of OBJ RAM, it actually needs 512 to prevent sprite blinking in some scenes. We have advanced a bit more with the schematics and it looks like the two RAMs I showed above are table double buffers applied for each line. The game parses the whole sprite table and select the sprites to draw in the next line and stores them in those RAMs, which then are parsed in the next line.
Long story short, it seems that the real number is 512 bytes, not 256 bytes. Could you please reopen this issue and give me a few more days to finish extracting the schematics?
I'm sorry again for having rushed about this.

**Comment by cuavas** [comment]
So are you saying it has a per-line sprite limit of 32 sprites?  Have you determined how it selects which sprites will be dropped if the limit is exceeded?

**Comment by jotego** [comment]
We finished extracting the schematics (find them attached). The sprite subsystem is at page 8. I have not simulated the circuit, but it looks to me like the first 32 sprites that match a line are the ones drawn. The logic is indeed as I described above, the whole sprite table is first parsed to find matches for the scan line + 2. Those matches are stored in a smaller RAM that can hold up to 32 sprites. That RAM is parsed in the next line and a line buffer is filled, which finally gets dumped in yet the next line. So the whole process takes three lines.

[kunio.pdf](https://github.com/mamedev/mame/files/10110750/kunio.pdf)

**Comment by cuavas** [comment]
Probably best to simulate properly with a scanline callback.
